classes/request.class.php

```diff
@@ -212,11 +212,22 @@ public function send_request($endpoint, $requestbody, $method, $requesttype = 'g
      */
     public function test_connection($apiurl, $apikey) {
 
+        $valid_url_regex = '/.+\.(turnitin\.com|turnitin\.co\.uk|turnitin\.dev|turnitin\.org|tii-sandbox\.com)\/api$/m';
```

Comment:
Can you rename the variable to $validurlregex please? Moodle code checker complains if variable have underscores. It only accepts lowercase names. Makes it less readable but it is what it is.